src/pages/LoginHandler/index.js

```diff
@@ -42,114 +40,104 @@ class LoginHandler extends React.Component {
     });
   };
 
-  enableSubmitBtn = () => {
-    this.setState({
-      variantVal: 'primary',
-      btnColor: 'white',
-      btnDisabled: 'false',
-    });
-  };
-
-  disableSubmitBtn = () => {
-    this.setState({
-      variantVal: 'tertiary',
-      btnColor: 'black',
-      btnDisabled: 'true',
-    });
-  };
-
-  handleUserNameInputChange = username => {
-    const { password } = this.state;
-    this.setState({
-      username,
-    });
-    if (password !== '' && username !== '') {
-      this.enableSubmitBtn();
-    } else {
-      this.disableSubmitBtn();
-    }
-  };
-
-  handlePassWordInputChange = password => {
-    const { username } = this.state;
-    this.setState({
-      password,
-    });
-    if (username !== '' && password !== '') {
-      this.enableSubmitBtn();
-    } else {
-      this.disableSubmitBtn();
-    }
-  };
-
-  handleLoginSubmit = () => {
-    const { username, password } = this.state;
-    const { dispatch } = this.props;
-    dispatch({
+  const handleLoginSubmit = async () => {
+    const { dispatch } = props;
+    const response = await dispatch({
       type: 'auth/loginUser',
       payload: {
         username,
         password,
       },
-    }).then(response => {
-      const { status } = response;
-      if (status === 'success') {
-        this.setLoggedIn({ username });
-      }
     });
+    const { status, message } = response;
+    setAlertComponent({
+      message,
+      variant: status === 'failure' ? 'danger' : 'success',
+    });
+    setAlertOpen(true);
+    if (status === 'success') {
+      // wait for a couple
+      // of seconds before taking the
+      // user to login page.
+      setTimeout(() => {
+        // set logged in to true
+        setLoggedIn({ username });
+      }, 2000);
```

Comment:
If login is successful, why are we waiting? And are we really taking the user to the login page on successful login?

Reply:
So, what I thought initially is to show an alert for a couple of seconds that the login attempt was successful, then redirect to the login page, to provide better user feedback, else as soon as the alert appears, the user is redirected.

Also, the function `setLoggedIn` pushes the `/private` page, which will be the Overview page after anisha's PR

Reply:
Speaking totally for myself, I'm not a fan of ephemeral effects on web pages.  If my attention happens to be pulled away while I'm waiting for the server to respond, I may miss the alert.  In this particular case, that's probably OK, since I'll presumably end up where I expected and wanted to be, anyway...but, that mostly serves to underscore the notion that displaying the alert is not important (and, therefore, perhaps not worth the complication).

I would recommend doing the redirect immediately and then putting a banner (or whatever it's called) on the target page which contains the alert -- that way, it's waiting there for the user when his/her attention returns, and s/he can dismiss the banner once s/he has read it.